### PR TITLE
fix(dlm): table structure synchronization failure when table names are inconsistent

### PR DESCRIPTION
**Pull request by guowl3**
#### What type of PR is this?
type-bug

#### What this PR does / why we need it:
The structure comparison module has added a target table name field, which is used to generate change scripts for tables with different names. After this PR is merged, the DLM module will support table structure synchronization for tables with different names.

#### Which issue(s) this PR fixes:

Fixes #[2460](https://github.com/oceanbase/odc/issues/2460)

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/structurecompare/comparedbobject/AbstractDBObjectStructureComparator.java [hunk 1]
```diff
@@ -32,10 +32,15 @@ public abstract class AbstractDBObjectStructureComparator<T extends DBObject>
         implements DBObjectStructureComparator<T> {
     protected final String srcSchemaName;
     protected final String tgtSchemaName;
+    protected final String srcTableName;
+    protected final String tgtTableName;
```
Comment (PeachThinking) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/structurecompare/comparedbobject/AbstractDBObjectStructureComparator.java:36]: It is recommended to change these 2 fields to：
`projceted final List<String> srcObjectNames;`
`projceted final List<String> tgtObjectNames;`

**Comment by PeachThinking** [comment]
👍👍👍～
could you please add some compare tables with different name in ut test like `DefaultDBStructureComparatorTest`

**Comment by guowl3** [comment]
> 👍👍👍～ could you please add some compare tables with different name in ut test like `DefaultDBStructureComparatorTest`
added unit test 👉 DBTableStructureComparatorTest.java

**Review by PeachThinking — APPROVED** [review]
lgtm

**Review by smallsheeeep — APPROVED** [review]

**Review by yizhouxw — APPROVED** [review]

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**